### ISSUE TITLE
Compilation errors on NANO 33 IoT (issue with QwI2C class)

### ISSUE THREAD

**Issue by cpsiff** [sparkfun/SparkFun_Qwiic_TMF882X_Arduino_Library#4]
When I try to compile one of the example programs, e.g. Example-01_Basic.ino, I get compilation errors related to the QwI2C class in this library.

Steps to reproduce:
1. Install SparkFun Qwiic TMF882X library via Arduino UI (bug appears on v1.0.0 and v1.0.1)
2. Create new sketch and copy in some example code
3. Select Arduino Nano 33 IoT as your board
4. Click the "verify" checkmark

Full error log:
```
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_tmf882x.h:53:0,
                 from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_tmf882x.cpp:52:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:15: error: 'TwoWire' has not been declared
     bool init(TwoWire& wirePort, bool bInit=false);
               ^~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:77:5: error: 'TwoWire' does not name a type; did you mean 'TwoWire_h'?
     TwoWire* _i2cPort;
     ^~~~~~~
     TwoWire_h
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_tmf882x.h:53:0,
                 from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/sfe_shim.cpp:62:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:15: error: 'TwoWire' has not been declared
     bool init(TwoWire& wirePort, bool bInit=false);
               ^~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:77:5: error: 'TwoWire' does not name a type; did you mean 'TwoWire_h'?
     TwoWire* _i2cPort;
     ^~~~~~~
     TwoWire_h
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:48:0:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:15: error: 'TwoWire' has not been declared
     bool init(TwoWire& wirePort, bool bInit=false);
               ^~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:77:5: error: 'TwoWire' does not name a type; did you mean 'TwoWire_h'?
     TwoWire* _i2cPort;
     ^~~~~~~
     TwoWire_h
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_tmf882x.h:53:0,
                 from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/sfe_arduino.cpp:60:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:15: error: 'TwoWire' has not been declared
     bool init(TwoWire& wirePort, bool bInit=false);
               ^~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:77:5: error: 'TwoWire' does not name a type; did you mean 'TwoWire_h'?
     TwoWire* _i2cPort;
     ^~~~~~~
     TwoWire_h
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In constructor 'QwI2C::QwI2C()':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:74:22: error: class 'QwI2C' does not have any field named '_i2cPort'
 QwI2C::QwI2C(void) : _i2cPort{nullptr}
                      ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: At global scope:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:83:6: error: prototype for 'bool QwI2C::init(arduino::TwoWire&, bool)' does not match any in class 'QwI2C'
 bool QwI2C::init(TwoWire &wirePort, bool bInit)
      ^~~~~
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:48:0:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:10: error: candidates are: bool QwI2C::init(int&, bool)
     bool init(TwoWire& wirePort, bool bInit=false);
          ^~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:63:10: error:                 bool QwI2C::init()
     bool init(void);
          ^~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In member function 'bool QwI2C::init()':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:103:10: error: '_i2cPort' was not declared in this scope
     if (!_i2cPort)
          ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:103:10: note: suggested alternative: 'Port'
     if (!_i2cPort)
          ^~~~~~~~
          Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:104:31: error: no matching function for call to 'QwI2C::init(arduino::TwoWire&, bool)'
         return init(Wire, true); // no wire, send in Wire and init it
                               ^
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:100:6: note: candidate: bool QwI2C::init()
 bool QwI2C::init(void)
      ^~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:100:6: note:   candidate expects 0 arguments, 2 provided
In file included from /home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:48:0:
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:10: note: candidate: bool QwI2C::init(int&, bool)
     bool init(TwoWire& wirePort, bool bInit=false);
          ^~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.h:64:10: note:   no known conversion for argument 1 from 'arduino::TwoWire' to 'int&'
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In member function 'bool QwI2C::ping(uint8_t)':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:116:10: error: '_i2cPort' was not declared in this scope
     if (!_i2cPort)
          ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:116:10: note: suggested alternative: 'Port'
     if (!_i2cPort)
          ^~~~~~~~
          Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:119:5: error: '_i2cPort' was not declared in this scope
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:119:5: note: suggested alternative: 'Port'
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
     Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In member function 'bool QwI2C::writeRegisterByte(uint8_t, uint8_t, uint8_t)':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:131:10: error: '_i2cPort' was not declared in this scope
     if (!_i2cPort)
          ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:131:10: note: suggested alternative: 'Port'
     if (!_i2cPort)
          ^~~~~~~~
          Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:134:5: error: '_i2cPort' was not declared in this scope
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:134:5: note: suggested alternative: 'Port'
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
     Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In member function 'int QwI2C::writeRegisterRegion(uint8_t, uint8_t, uint8_t*, uint16_t)':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:166:5: error: '_i2cPort' was not declared in this scope
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:166:5: note: suggested alternative: 'Port'
     _i2cPort->beginTransmission(i2c_address);
     ^~~~~~~~
     Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp: In member function 'int QwI2C::readRegisterRegion(uint8_t, uint8_t, uint8_t*, uint16_t)':
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:186:10: error: '_i2cPort' was not declared in this scope
     if (!_i2cPort)
          ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:186:10: note: suggested alternative: 'Port'
     if (!_i2cPort)
          ^~~~~~~~
          Port
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:202:9: error: '_i2cPort' was not declared in this scope
         _i2cPort->beginTransmission(addr);
         ^~~~~~~~
/home/carter/Arduino/libraries/SparkFun_Qwiic_TMF882X_Library/src/qwiic_i2c.cpp:202:9: note: suggested alternative: 'Port'
         _i2cPort->beginTransmission(addr);
         ^~~~~~~~
         Port

exit status 1

Compilation error: exit status 1
```

**Comment by PaulZC** [comment]
Hi Carter (@cpsiff ),

Have you included Wire?

```
#include <Wire.h>
```

Best,
Paul

**Comment by PaulZC** [comment]
Humm. It looks like it should be included automatically though...

Maybe try including Wire before the TMF library header?

```
#include <Wire.h>
#include "SparkFun_TMF882X_Library.h"
```

**Comment by PaulZC** [comment]
That didn't fix it. Looks like [we are not alone](https://www.google.com/search?q=Arduino+NANO+33+TwoWire)...

**Comment by PaulZC** [comment]
Hi Carter (@cpsiff ),

OK. I have a solution...

If you edit Wire.h and insert the line ```#include "Arduino.h"``` at the start, the code compiles successfully.

On my Windows machine, Wire.h is in:

```
C:\Users\<my_user>\AppData\Local\Arduino15\packages\arduino\hardware\samd\1.8.13\libraries\Wire\Wire.h
```

At the start of the file, add  ```#include "Arduino.h"``` next to the other includes:

```
#ifndef TwoWire_h
#define TwoWire_h

#include "api/HardwareI2C.h"
#include "variant.h"
#include "SERCOM.h"
#include "Arduino.h" // ADD ME
```

I think the main issue is to do with namespaces, and class TwoWire being inside namespace arduino?

Let us know if this solves your issue.

Best wishes,
Paul

**Comment by PaulZC** [comment]
Hi Carter (@cpsiff ),

Thanks for reporting. This is resolved in v1.0.2:

https://github.com/sparkfun/SparkFun_Qwiic_TMF882X_Arduino_Library/releases/tag/v1.0.2

Best wishes,
Paul
